airflow/utils/log/logging_mixin.py

```diff
@@ -179,15 +182,15 @@ def set_context(logger, value):
     :param logger: logger
     :param value: value to set
     """
-    _logger = logger
-    while _logger:
-        for handler in _logger.handlers:
+    while logger:
+        for handler in logger.handlers:
             # Not all handlers need to have context passed in so we ignore
             # the error when handlers do not have set_context defined.
             set_context = getattr(handler, 'set_context', None)
             if set_context:
-                set_context(value)
-        if _logger.propagate is True:
-            _logger = _logger.parent
+                if set_context(value) is DISABLE_PROPOGATE:
```

Comment:
Should we just chain this "if" with the "if" above?

```python
if set_context and set_context(value) is DISABLE_PROPOGATE:
```

Reply:
Can do, sure

Reply:
Generally the visibility of warnings may be added into cfg as a section like deprecation or dependency warnings so that we can able to path around them.